instrumentation/aws-lambda-1.0/library/src/main/java/io/opentelemetry/instrumentation/awslambda/v1_0/ParentContextExtractor.java

```diff
@@ -5,35 +5,28 @@
 
 package io.opentelemetry.instrumentation.awslambda.v1_0;
 
-import com.amazonaws.serverless.proxy.model.Headers;
 import io.opentelemetry.api.OpenTelemetry;
 import io.opentelemetry.context.Context;
 import io.opentelemetry.context.propagation.TextMapPropagator;
 import java.util.Collections;
 import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
 
 public class ParentContextExtractor {
 
-  static Context fromHttpHeaders(Headers headers) {
-
+  static Context fromHttpHeaders(Map<String, String> headers) {
     return OpenTelemetry.getGlobalPropagators()
         .getTextMapPropagator()
-        .extract(io.opentelemetry.context.Context.current(), headers, HeadersGetter.INSTANCE);
+        .extract(
+            io.opentelemetry.context.Context.current(), lowercaseMap(headers), MapGetter.INSTANCE);
   }
 
-  private static class HeadersGetter implements TextMapPropagator.Getter<Headers> {
-
-    private static final HeadersGetter INSTANCE = new HeadersGetter();
-
-    @Override
-    public Iterable<String> keys(Headers map) {
-      return map.keySet();
-    }
-
-    @Override
-    public String get(Headers headers, String s) {
-      return headers.getFirst(s);
-    }
+  private static Map<String, String> lowercaseMap(Map<String, String> source) {
+    return source.entrySet().stream()
+        .collect(
+            Collectors.toMap(
+                e -> e.getKey() == null ? null : e.getKey().toLowerCase(), Entry::getValue));
```

Comment:
`e -> e.getKey() == null ? null`

:O:O:O
Wow

Reply:
What do you think about `.filter(e -> e.getKey() != null)`?

Reply:
The idea here was a typical one - some map implementations allow NULLed keys. But since it's about HTTP headers, filtering out makes sense. Thanks :)